### ISSUE TITLE
[from:data] Implement unfrozen encoder for Laser L D10k/C20k

### ISSUE THREAD

**Issue by kaihe** [kaihe/contra_nes_policy#22]
## Summary
Implement an end-to-end pixel training path to test whether fine-tuning the existing static image encoder improves the Laser L-policy result at the exact D10k/C20k recipe.

## Done / request
- [ ] Add a pixel-backed Laser datahouse loader; precomputed tokens cannot support encoder gradients.
- [ ] Allow the existing pretrained static encoder to remain trainable with the L core.
- [ ] Support separate optimizer learning rates: policy core 1e-4 and encoder 1e-5 initially.
- [ ] Preserve effective batch exposure, using gradient accumulation if raw-image VRAM reduces the physical batch.
- [ ] Run a 200-500 step throughput and memory smoke test before the full experiment.
- [ ] Train and publish L / Laser-only D10k / C20k / seed 0 with checkpoints near 2k, 5k, 10k and final 20k.

## Paths
- Current frozen Laser config: /home/kaihe/code/contra_nes_policy/src/contra_policy/config_bc_laser.yaml
- Current trainer guard: /home/kaihe/code/contra_nes_policy/src/contra_policy/train_bc.py
- Existing encoder model: /home/kaihe/code/contra_nes_policy/src/contra_policy/model.py
- Laser datahouse: /home/kaihe/code/contra_nes_data/game_trace/datahouse/level1/boss/laser/
- Encoder comparison: /home/kaihe/code/contra_nes_evaluation/doc/0029-result-laser-encoder-comparison.md

## Next action (for this repo)
1. Reuse or generalize the frame-loader work tracked in issue #19 for Laser raw-frame episodes.
2. Add a proposed experiment doc and implement pixel-backed encoder fine-tuning with differential learning rates.
3. Verify the smoke run, then execute the exact L/D10k/C20k cell and hand its checkpoints to evaluation.
4. Done when evaluation can compare it directly against frozen static 19% and frozen temporal 17% under the 0029 protocol.

## Must not break
- Change only encoder trainability in the comparison: keep Laser-only D10k, C20k, L shape, lr/schedule exposure, seed and evaluation protocol matched.
- Do not silently consume cached tokens when the encoder is trainable.
- Preserve action/frame causal alignment and the existing uid-digest validation split.
- Avoid a shared 1e-4 rate on the pretrained encoder; expose an encoder-specific rate.

## Related
- kaihe/contra_nes_policy#19 provides the existing raw-frame loader request and alignment contract.
- Policy doc/0023-exp-frame-difference-laser-policy.md
- Evaluation doc/0029-result-laser-encoder-comparison.md

**Comment by kaihe** [comment]
Scheduled on  as a matched goal-free pair: frozen control, then encoder-unfrozen candidate (core LR 1e-4, encoder LR 1e-5). Raw-frame membership verified at 9,771 train episodes; both GPU smoke runs and the full policy test suite pass.

**Comment by kaihe** [comment]
Data handoff is complete (`contra_nes_data#15`, commit `a758342`, branch `feat/laser-reduced-features`).

Use catalog selector:
```json
{"level":1,"task":"boss","weapon":"laser","representation":"reduced-view-v1","encoder_sha256":"f36041bc69f1ce20781d5200bc89970b1b305e12bff5ae826b23581ca0f1923c"}
```

Local release spec (complete ordered membership and all shard hashes):
`/home/kaihe/code/contra_nes_data/game_trace/datahouse/level1/boss/laser/features/reduced-view-v1/f36041bc69f1ce20781d5200bc89970b1b305e12bff5ae826b23581ca0f1923c/spec.json`

Published: 42 shards / 10,293 episodes / 1,075,404 frames / 8,856,750,080 bytes. Features are float16 `(T,256,4,4)` after frozen `view_backbone + reduce`; `proj` and `token_ln` are excluded. Raw action arrays are preserved and policy must retain target shift `actions[i+1]`.
